fontc/src/lib.rs

```diff
@@ -148,17 +145,18 @@ fn add_finalize_static_metadata_ir_job(
                     | AnyWorkId::Fe(FeWorkIdentifier::FinalizeStaticMetadata)
             )
         });
+
         workload.insert(
-            FeWorkIdentifier::FinalizeStaticMetadata.into(),
+            work.id().into(),
             Job {
-                work: create_finalize_static_metadata_work().into(),
+                work: work.into(),
                 dependencies,
                 read_access: ReadAccess::Custom(read_access),
                 write_access,
             },
         );
     } else {
-        workload.mark_success(FeWorkIdentifier::FinalizeStaticMetadata);
+        workload.mark_success(work.id());
```

Comment:
cool, this code is now starting to look much more general which suggests maybe we can leverage generics & traits more, going forward.